src/backend/utils/adt/ruleutils.c

```diff
@@ -3273,7 +3273,18 @@ print_function_arguments(StringInfo buf, HeapTuple proctup,
 		if (argname && argname[0])
 			appendStringInfo(buf, "%s ", quote_identifier(argname));
 		appendStringInfoString(buf, format_type_be(argtype));
-		if (print_defaults && isinput && inputargno > nlackdefaults)
+
+		if (nextargdefault != NULL && IsA(lfirst(nextargdefault), FuncDefault))
```

Comment:
Just wondering which test case is failing without this?

Reply:
Yeah looks like this is already getting handled in `get_rule_expr` and the above logic can be removed.